src/georaster-layer-for-leaflet.ts

```diff
@@ -224,6 +227,16 @@ const GeoRasterLayer: (new (options: GeoRasterLayerOptions) => any) & typeof L.C
     }
   },
 
+  onAdd: function (map) {
+    if (!this.options.maxZoom) {
+      // maxZoom is needed to display the tiles in the correct order over the zIndex between the zoom levels
+      // https://github.com/Leaflet/Leaflet/blob/2592967aa6bd392db0db9e58dab840054e2aa291/src/layer/tile/GridLayer.js#L375C21-L375C21
+      this.options.maxZoom = map.getMaxZoom();
+    }
```

Comment:
This is needed because else no zIndex is applied to the Tilelayer in the DOM and then it takes the natural order (top element is visible). With caching it turned out, that now rerendering is **not** triggered and then the canvas / tile from the zoom before is shown.
With adding a maxZoom, it applies a zIndex. Instead of this lines, we can add a value to the `maxZoom` option, like TileLayer from Leaflet.

![grafik](https://github.com/GeoTIFF/georaster-layer-for-leaflet/assets/19800037/e28415bd-f771-48d2-b3ae-aee4262984fe)

Can be reproduced with:
1. Open test file caching.html
2. run `Object.values(map._layers).find(x => x instanceof GeoRasterLayer).options.maxZoom = undefined` in the console
3. Click now **once** on "-" of the zoom buttons
4. You are now on zoom 16 (high resolution)
5. Click now **once again** on "-" of the zoom buttons
6. You are now on zoom 15 (low resolution)
7. Click now **once** on "+" of the zoom buttons
8. You are now on zoom 16 (**but** it has low resolution!) -> This is because of the missing zIndex (maxZoom)

Current:
![maxzoom-zindex-wrong](https://github.com/GeoTIFF/georaster-layer-for-leaflet/assets/19800037/f4e2a54f-3063-457a-8c48-4b0f640baf89)

Fixed:
![maxzoom-zindex](https://github.com/GeoTIFF/georaster-layer-for-leaflet/assets/19800037/1543ed18-8215-4c32-897c-9c156b208cc9)

Reply:
wow.  that's a great catch!